### PR TITLE
What's New in Accessibility in the .NET Framework 

### PR DESCRIPTION
**Pull request by rpetrusha**
# What's New in Accessibility in the .NET Framework 

## Summary

Since accessibility is a major focus, it makes sense to add a separate what's new topic for it.

The topic is published on our review site at https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility?branch=pr-en-us-3522.

## Suggested Reviewers
@preetikr, @vivmishra, @Tanya-Solyanik  , @merriemcgaw

### CODE REVIEW

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+- [Windows Presentation Foundation (WPF)](windows-presentation-foundation-wpf)
+
+- [Windows Forms](windows-forms-accessibility-improvements)
+
+### Windows Presentation Foundation (WPF)
+
+**Screen reader improvements**
+
+If accessibility improvcements are enabled, the .NET Framework 4.7.1 includes the following enhancements that affect screen readers:
```
Comment (Tanya-Solyanik) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: typo

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+```xml
+<runtime>
+    <!-- AppContextSwitchOverrides value attribute is in the form of 'key1=true|false;key2=true|false  -->
+    <AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false" />
+</runtime>
+```
+
+Similarly, applications that target versions of the .NET Framework starting with 4.7.1 can disable accessibility features adding the following switch to the [`<AppContextSwitchOverrides>`](~/docs/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) element in the [`<runtime>`](~/docs/configure-apps/file-schema/runtime/index.md) section of the application's configuration file.
```
Comment (Tanya-Solyanik) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: should it be "by adding"?

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    Before:
+
+    ![Disabled text before accessibility improvements](media/wf-disabled-before.png) 
+
+    After:
+
+    ![Disabled text after accessibility improvements](media/wf-disabled-after.png) 
+- High contrast improvements in the Thread Exception Dialog.
+
+- Better multi-monitor DPI awareness in control anchoring and scaling.
```
Comment (Tanya-Solyanik) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: This item is not a part of accessibility work

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+- The <xref:System.Windows.Forms.CheckedListBox> control notifies Narrator when the <xref:System.Windows.Forms.CheckedListBox.CheckedState> property has changed so the user is notified that they’ve changed the value of a list item.
+ 
+- The <xref:System.Windows.Forms.DataGridViewCell> control reports the correct read-only status to Narrator.
+ 
+- Narrator can now read disabled <xref:System.Windows.Forms.ToolStripMenuItem> text, whereas previously it would skip over disabled menu items.
+
+**Enhanced UI accessibility patterns**
+
+Starting with the .NET Framework 4.7.1, developers of accessibility technology tools can leverage common api accessibility patterns and properties for several WinForms controls. These accessibility improvements include:
```
Comment (Tanya-Solyanik) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: I thought that API is usually spelled in all-caps, no?

Comment (Tanya-Solyanik) [reply docs/framework/whats-new/whats-new-in-accessibility.md]: it should be "Enhanced support for UIAutomation accessibility patterns" - this is how WPF calls UIA in the same doc

**Review by Tanya-Solyanik — CHANGES_REQUESTED** [review]

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+  - "what's new [.NET Framework]"
+author: "rpetrusha"
+ms.author: "ronpet"
+manager: "wpickett"
+---
+
+# What's new in accessibility in the .NET Framework
+
+The .NET Framework aims at making applications more accessibile for your users. Accessibility features allow an application to handle a range of devices and form factors seamlessly. Starting with the .NET Framework 4.7.1, the .NET Framework includes a large number of accessibility improvements that allow developers to create accessible applications. 
+
```
Comment (merriemcgaw) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: Could you say something about supporting Assistive Technology apps here?  It's not so much about form factors as about providing the right experience for users of assistive technology.

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+The new accessibility features are enabled by default for applications that target the .NET Framework 4.7.1 or later. In addition, applications that target an earlier version of the .NET Framework but are running on the .NET Framework 4.7.1 or later can opt out of legacy accessibility behaviors by adding the following switch to the [`<AppContextSwitchOverrides>`](~/docs/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) element in the [`<runtime>`](~/docs/configure-apps/file-schema/runtime/index.md) section of the application's configuration file.
```
Comment (merriemcgaw) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: This isn't super clear that opting out of legacy behaviors is the same thing as opting in to the 4.7.1 improvements.

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+- Disabled text is easier to read. For example:
+
+    Before:
+
+    ![Disabled text before accessibility improvements](media/wf-disabled-before.png) 
+
+    After:
+
+    ![Disabled text after accessibility improvements](media/wf-disabled-after.png) 
+- High contrast improvements in the Thread Exception Dialog.
```
Comment (merriemcgaw) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: Would a before/after image be helpful here?

File: docs/framework/whats-new/whats-new-in-accessibility.md (hunk 4 above)
Comment (merriemcgaw) [reply docs/framework/whats-new/whats-new-in-accessibility.md]: Agree with @Tanya-Solyanik - "Enhanced support for UIAutomation accessibility patterns" sounds much better.

File: docs/framework/whats-new/whats-new-in-accessibility.md (hunk 7 above)
Comment (rpetrusha) [reply docs/framework/whats-new/whats-new-in-accessibility.md]: There is a before/after image here. Sorry, @merriemcgaw  -- I should have included a link to the staged published topic. It is [What's new in accessibility in the .NET Framework](https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility?branch=pr-en-us-3522).

**Comment by rpetrusha** [comment]
@Tanya-Solyanik and @merriemcgaw, thanks for your review. I've addressed all of your comments. When the build finishes, the staged topic is available [here](https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility?branch=pr-en-us-3522).

**Comment by BillWagner** [comment]
@rpetrusha Can you fix the merge conflicts here? Then, I'll give it a review.

**Comment by rpetrusha** [comment]
Thanks, @BillWagner. I've resolved the merge conflict.

**Comment by JRAlexander** [comment]
I think you need to make this MVC compliant.

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+  - <xref:System.Windows.Automation.AutomationLiveSetting.Assertive?displayProperty=nameWithType>. The element sends interruptive notifications if the content of the live region has changed.   
+
+You can create a LiveRegion by setting the **AutomationProperties.LiveSetting** property on the element of interest, as shown in the following example:   
+
+```xaml
+<TextBlock Name="myTextBlock" AutomationProperties.LiveSetting="Assertive">announcement</TextBlock>
+```
+
+When the data in the live region changes and you need to inform a screen reader, you explicitly raise an event, as shown in the following sample.
+
```
Comment (BillWagner) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: Should these be in tabs?

File: docs/framework/whats-new/whats-new-in-accessibility.md [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+```vb
+Dim peer = FrameworkElementAutomationPeer.FromElement(myTextBlock)
+peer.RaiseAutomationEvent(AutomationEvents.LiveRegionChanged)
+
+```
+
+**High contrast**
+
+Starting with the .NET Framework 4.7.1, improvements in high conrast have been made to various WPF controls. They are now visible when the <xref:System.Windows.SystemParameters.HighContrast%2A> theme is set. These include:
```
Comment (BillWagner) [inline docs/framework/whats-new/whats-new-in-accessibility.md]: typo: conrast => contrast.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this looks great.

I found 1 typo, and had one suggestion.

Also, this new article isn't in the TOC. Do you want to add it, or only have it available from the main "What's new" topic?

File: docs/framework/whats-new/whats-new-in-accessibility.md (hunk 9 above)
Comment (rpetrusha) [reply docs/framework/whats-new/whats-new-in-accessibility.md]: Thanks, @BillWagner. I'll add it to the TOC.

**Comment by rpetrusha** [comment]
I've addressed your comments, @BillWagner. The accessibility topic already was in the TOC as a child of the What's New in the .NET Framework topic. Should it be a peer rather than a child?

**Comment by mairaw** [comment]
I think it's fine as a child like it's showing on review now.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
@rpetrusha It's fine. I reviewed the diffs, and I thought it was a new file, but I didn't see a TOC change. I say :shipit: when ready.

**Pull request merged**